### PR TITLE
SPOI-9320 HDFS sync appHub app

### PR DESCRIPTION
**Pull request by yogidevendra**

### CODE REVIEW

File: tutorials/hdfs-sync/src/main/java/com/datatorrent/apps/copy/HDFSFileCopyApp.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (DT-Priyanka) [inline tutorials/hdfs-sync/src/main/java/com/datatorrent/apps/copy/HDFSFileCopyApp.java]: Do we want to put DataTorrent license here?

File: tutorials/hdfs-sync/src/site/conf/cluster-memory-conf.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+<configuration>
```
Comment (DT-Priyanka) [inline tutorials/hdfs-sync/src/site/conf/cluster-memory-conf.xml]: can you explain what cluster-memory-config is? This looks more like operator memory config

File: tutorials/hdfs-sync/src/site/conf/sandbox-memory-conf.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+<configuration>
+  <property>
+    <name>dt.operator.HDFSInputModule.prop.maxReaders</name>
```
Comment (DT-Priyanka) [inline tutorials/hdfs-sync/src/site/conf/sandbox-memory-conf.xml]: same why do we have operator memory properties in sandbox-memory-conf ?

File: tutorials/hdfs-sync/src/main/java/com/datatorrent/apps/copy/HDFSFileCopyApp.java (hunk 1 above)
Comment (yogidevendra) [inline tutorials/hdfs-sync/src/main/java/com/datatorrent/apps/copy/HDFSFileCopyApp.java]: This code is not part of Apache project. But, this is open-sourced under Apache license for any one to use, make changes etc.
This is approved by Amol, Ashwin.

File: tutorials/hdfs-sync/src/site/conf/cluster-memory-conf.xml (hunk 2 above)
Comment (yogidevendra) [inline tutorials/hdfs-sync/src/site/conf/cluster-memory-conf.xml]: Currently, launch UI has no way to separate out mandatory properties and advanced properties.
Thus, as a workaround we have the following strategy:
1. All the properties for which end user must specify the values (e.g. input path, output path etc.) are part of properties.xml.
2. All other properties are (for which we can set reasonable defaults) are part of apconf.
3. apconf have two profiles: 
    a. cluster which is suitable for medium to big cluster deployment.
    b. sandbox which is suitable for RTS sanbox / single node deployment.

File: tutorials/hdfs-sync/src/site/conf/sandbox-memory-conf.xml (hunk 3 above)
Comment (yogidevendra) [inline tutorials/hdfs-sync/src/site/conf/sandbox-memory-conf.xml]: To minimize memory footprint in the sandbox.

**Pull request merged**